rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -14,32 +14,48 @@ use uuid::Uuid;
 use zbus::zvariant::{self, Value};
 
 const ETHERNET_KEY: &str = "802-3-ethernet";
+const BONDING_KEY: &str = "bond";
 const WIRELESS_KEY: &str = "802-11-wireless";
 const WIRELESS_SECURITY_KEY: &str = "802-11-wireless-security";
 const LOOPBACK_KEY: &str = "loopback";
 
+
+
 /// Converts a connection struct into a HashMap that can be sent over D-Bus.
 ///
+/// * `net`: The full NetworkState -- used for correlations between interfaces.
 /// * `conn`: Connection to convert.
-pub fn connection_to_dbus(conn: &Connection) -> NestedHash {
+pub fn connection_to_dbus<'a>(net: &'a NetworkState, conn: &'a Connection) -> NestedHash<'a>{
     let mut result = NestedHash::new();
     let mut connection_dbus = HashMap::from([
         ("id", conn.id().into()),
         ("type", ETHERNET_KEY.into()),
         ("interface-name", conn.interface().into()),
     ]);
+    if let Some(master) = &conn.base().master {
+        connection_dbus.insert("master", master.into());
+        connection_dbus.insert("slave-type", find_slave_type(net, master).unwrap().into());
```

Comment:
I'm not sure if we should gather the `slave-type` here or we should also store it in the `network::model::Connection` as well. The benefit would be, that we do not need to pass the `NetworkState` here.

Reply:
Move slave-type to model.rs and rename to port